svo (refiller).xml

```diff
@@ -812,38 +514,21 @@ function svo.rf_nextpotion()
     else
       svo.rf_currenttrans = svo.rf_transtoxicology
     end
-
-    svo.rf_fillpot(svo.rf_refilling.currentorder,
-      -- refill 5 health 2 arty means 3 normal + 2 arty!
-      (svo.rf_refilling.p[svo.rf_refilling.currentorder].normal - svo.rf_refilling.p[svo.rf_refilling.currentorder].arty) * (svo.rf_currenttrans and 4 or 5) +
-      svo.rf_refilling.p[svo.rf_refilling.currentorder].arty * 2)
-    svo.rf_boilpot()
-
-    -- clear the fill so next time rf_nextpotion() is called, it's fine
-    svo.rf_refilling.p[svo.rf_refilling.currentorder] = nil
-  end
-end
-
-function svo.rf_undopot(potion, fills, pot)
-  svo.assert(potion and fills, "rf_undopot: need to supply potion and how many fills went in")
-
-  svo.assert(concoctions[potion], "rf_undopot: don't know about such a potion")
-  for item, amount in pairs(concoctions[potion]) do
-    svo.doadd("get "..amount * fills.." "..item.." from "..(pot or conf.potid))
+    svo.rf_brewpot()
+    -- if we have made all of the current item, then clear the fill so next time rf_nextpotion() is called, it is fine
+    max = (svo.rf_currenttrans and 12 or 10)
+    if svo.rf_refilling.currentorderdata &lt;= max then
+      svo.rf_refilling.p[svo.rf_refilling.currentorder] = nil
+--      svo.rf_refilling.currentorderdata = nil
+      svo.rf_refilling.currentorder = nil
+    end
   end
-  svo.doadd("get "..(pot or conf.potid))
-  svo.doadd("refill next potion")
 end
 
 function svo.rf_missingstuff()
   disableTrigger("Missing ingredients"); tempTimer(2, function() enableTrigger("Missing ingredients") end)
-  killTimer(svo.rf_wait_to_boil)
-
-  echo'\n' svo.echof("Ack, looks like you're out of enough ingredients - going to get what we put into the pot back...")
 
-    svo.rf_undopot(svo.rf_refilling.currentorder,
-      (svo.rf_refilling.currentorderdata.normal - svo.rf_refilling.currentorderdata.arty) * (svo.rf_currenttrans and 4 or 5) +
-      svo.rf_refilling.currentorderdata.arty * 2)
+  echo'\n' svo.echof("Ack, looks like you're out of enough ingredients. :(")
```

Comment:
Dont forget to do a cancel here - dont need to keep trying on balance after it stops here.

Reply:
Cancel added